File: chain_of_responsibility.py
import math

class Request():
 def __init__(self, amount):
  print("Requested: ${}".format(amount))
  self.amount = amount

 def get(self, bill):
  count = math.floor(self.amount/bill)
  self.amount -= count * bill
  print("Dispense {} ${} bills".format(count,bill))
  return self

def main():
	req = Request(378)
	req.get(100).get(50).get(20).get(10).get(5).get(1)


if __name__ == '__main__':
	main()

File: chain_of_responsibility.java
class chain_of_responsibility {
    static class Request {
        private int amount = 0;

        Request(int amount) {
            System.out.println("Requested: $" + amount);
            this.amount = amount;
        }

        public Request get(int bill) {
            int count = (int) Math.floor(this.amount/bill);
            this.amount -= count * bill;
            System.out.println(String.format("Dispense %d $%d bill",count,bill));
            return this;
        }
    }

    public static void main(String[] args) {
        Request request = new Request(378);
        request.get(100).get(50).get(20).get(10).get(5).get(1);
    }
}

